### PR TITLE
doc: retooling to allow for docs build

### LINKED ISSUE #135: Documentation layout and proposal
This is a placeholder issue for getting documentation written.

Proposed layout is a simple set of markdown docs:

```
doc/
    introduction.md
    spec/
        api.md
        token.md
    registry/
        introduction.md
        configuration.md
        notification.md
        ...
```

Goals are as follows:

1. Everything is simple markdown. We enforce no builder.
2. Doc directory is user oriented (spec, operating, etc.
3. Developer documentation belongs in README.md + CONTRIBUTING.md.
4. Docs should be importable elsewhere.

### PR DESCRIPTION
**Pull request by moxiegirl**
Adding docs build to the Makefile
Adding files in support of Docs build.
This allows for `make docs` locally and also for pulling into the larger `docs.docker.com`

Signed-off-by: Mary Anthony <mary@docker.com>

### CODE REVIEW

**Comment by stevvooe** [comment]
Does this conflict with #245?

**Comment by moxiegirl** [comment]
@stevvooe Sven's changes are in his branch which is pretty old. I've pulled in your comments from the ticket. I'm also expecting to go in another direction for the .yml file.   So, the content changes I have here in the .yml file and the .md files are just placeholders; look for the real content changes to come in a later PR.

**Comment by moxiegirl** [comment]
yaaargh..I have to correct the lousy .gitignore sorry sec

File: Makefile [hunk 1]
```diff
@@ -61,3 +62,21 @@ binaries: ${PREFIX}/bin/registry ${PREFIX}/bin/registry-api-descriptor-template
...
 	@echo "+ $@"
 	@rm -rf "${PREFIX}/bin/registry" "${PREFIX}/bin/registry-api-descriptor-template"
+
+	
+# Use the existing docs build cmds from docker/docker
+# Later, we will move this into an import
+DOCS_MOUNT := $(if $(DOCSDIR),-v $(CURDIR)/$(DOCSDIR):/$(DOCSDIR))
+DOCSPORT := 8000
+GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD 2>/dev/null)
+DOCKER_DOCS_IMAGE := dhe-docs$(if $(GIT_BRANCH),:$(GIT_BRANCH))
```
Comment (stevvooe) [inline Makefile]: Does this have anything to do with DHE? Also, should this use the VERSION variable from above?

**Comment by moxiegirl** [comment]
The VERSION is set off of the docs/Dockerfile.  It is also kind of moot as we don't publish sites for the individual documentation.  We have tooling here so you can build locally and test when you are writing.

**Comment by dmp42** [comment]
Thanks @moxiegirl !
Do we have to rename doc to docs?

As far as I understand, only the Dockerfile requires this - and it can probably be easily modified to use an ENV var defaulting to docs fed through the build command.

**Comment by moxiegirl** [comment]
@dmp42 We are trying to keep our tooling uniform across quite a few repositories.  All the other repositories are using `docs` --- it seems reasonable to change this one rather than all the others.

**Comment by fredlf** [comment]
I'm not exactly sure what this does, why it replaces Sven's work, or what the new IA is that it seems to be planning for. What is the goal here? Can you please provide us with some information as to your plans and intent?

File: Makefile [hunk 2]
```diff
@@ -61,3 +62,21 @@ binaries: ${PREFIX}/bin/registry ${PREFIX}/bin/registry-api-descriptor-template
 clean:
 	@echo "+ $@"
 	@rm -rf "${PREFIX}/bin/registry" "${PREFIX}/bin/registry-api-descriptor-template"
+
+	
+# Use the existing docs build cmds from docker/docker
+# Later, we will move this into an import
+DOCS_MOUNT := $(if $(DOCSDIR),-v $(CURDIR)/$(DOCSDIR):/$(DOCSDIR))
+DOCSPORT := 8000
+GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD 2>/dev/null)
```
Comment (stevvooe) [inline Makefile]: The following two lines:

```
GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD 2>/dev/null)
DOCKER_DOCS_IMAGE := docker-docs$(if $(GIT_BRANCH),:$(GIT_BRANCH))
```

May be replaced with this:

```
DOCKER_DOCS_IMAGE := docker-docs-$(VERSION)
```

The branch name may not be very interesting to keep the image name unique. If the goal is for this to have local branch name (ie "master"), then please disregard.

**Comment by stevvooe** [comment]
LGTM with the small clarification around the use of git branch vs version.

**Comment by stevvooe** [comment]
@moxiegirl Could you rebase this branch so we can merge it? Let me know if you need help.

**Comment by moxiegirl** [comment]
@stevvooe rebased.

**Comment by moxiegirl** [comment]
LGTM

**Comment by fredlf** [comment]
My questions are still outstanding. I'm just trying to understand what this does.

**Comment by moxiegirl** [comment]
@fredlf This just carries Sven's changes. He was on vaca or got caught up in other stuff and hadn't finished out the other PR.  I basically ported Steve's comments to this one.

**Comment by fredlf** [comment]
@moxiegirl Well, okay. I thought his PR was complete, but I could easily be mistaken. What about the bits in here that seem to be IA changes adding a new "distribution" section? Am I misunderstanding that?

**Comment by moxiegirl** [comment]
@fredlf Sven's PR was 22 days old when I opened mine.  I made changes on a branch and tested it --- works end-to-end.  If there are things left undone on Sven's that are key, he should make a new PR when he gets back. I'm working the documentation for the 4/7 release with the team.

**Pull request merged**